### ISSUE TITLE
I am trying to reconnect to the ZDCChat but i cannot

### ISSUE THREAD

**Issue by Lawati97** [zendesk/zendesk_sdk_chat_ios#221]
- SDK version: 1.9.2.2
- Xcode version: 9.3
- iOS or Simulator: IOS
- iOS Version: 12.1
- Swift version: 4.2.1

Steps to Reproduce:

1.  I had set the ViewController to ZDCChatViewController
2. set the chatUI as active

So this is what happening whenever i start a chat it will normally do (since i have ZDCchat.start()), but then when i end chat and go to the ZDCChatViewController. It goes back to the prechatform but when i click on next it would say that the agents are offline and i am not able to reconnect to the chat.

**Comment by RonanMcH** [comment]
HI @Lawati97 ,
Could you please clarify what version of the ChatSDK you are on? We don't have a version 1.9.2.2 of the ChatSDK but we do of the SupportSDK so perhaps you mixed them up.

I am a little bit unclear as to the steps to recreate this issue, I have tried to do so but can't. Could you try to share some code of exactly how you are presenting the chat and detail the behaviours a little bit?

You should not need to reassign the `ZDCChatViewController` but always just use the following methods to [start a chat](https://developer.zendesk.com/embeddables/docs/ios-chat-sdk/chat#starting-a-chat-in-a-navigation-controller).
Thanks,
Ronan.

**Comment by Lawati97** [comment]
@ronan-mch I apologize i meant SDK Version 1.4.0, so this is what i am trying to do is that as soon you click on a button in the navigation controller it would take you to would directly take you ZDCChat but if i ended the chat and clicked on the same button in navigation Controller the ZDCChat won't show up again.

**Comment by Lawati97** [comment]
@ronan-mch so let me rephrase it s this was our solution we had a button and whenever you click it. It would take you to the pre chat form and then to the live chat and when you end it, it would take you back to the first page and we don't want that we want it to go back to the pre chat form

https://drive.google.com/open?id=1D_GQJa8AU8sRA4cXh94-KG1gB4F2M9h-

The link will take you to the video to demonstrate my current issue

Hussain Al Lawati

**Comment by HamstringAssassin** [comment]
Thanks for the video @Lawati97! 

I think I understand what the issue is - unfortunately, once a chat ends, the product behavior is to dismiss. If you need a custom UI with custom behavior, there are provider methods to help build your own. 

Thanks, 
Al

**Comment by Lawati97** [comment]
@HamstringAssassin thank you, would you guide me where can i find these information?

Thanks again,
Hussain Al Lawati

**Comment by danthedrummer** [comment]
Hi,

Here's the docs describing how to use the Chat API providers to build your own UI:
https://developer.zendesk.com/embeddables/docs/ios-chat-sdk/sessionapi

Best of luck!

**Comment by HamstringAssassin** [comment]
Hi @Lawati97, 

If there's anything else I can help with please feel free to get in touch - otherwise, we'll close this issue. 

Thanks again for getting in contact. 
Al
